### DISCUSSION TITLE
"untyped" SQL function parameters

### DISCUSSION THREAD

**Discussion by rvalek** [EvgSkv/logica#154] [Q&A] [answered]
Hey, I was trying to use SQL's [TIMESTAMP_DIFF](https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp_diff), but I can't pass in the third argument from Logica:
`TimestampDiff(CurrentTimestamp(), date, "DAY"); # in SQL DAY should be w/o quotes`
I get the following error from BQ:
`A valid date part name is required`

Is there a way to do this?

Thank you!

**Comment by EvgSkv** [comment, marked as answer]
We have to use `SqlExpr` to pass this.
So we should do something like this:
```
Q() = TimestampDiff(CurrentTimestamp(), CurrentTimestamp(), SqlExpr("DAY", {}));
```

or this:
```
Q() = SqlExpr("Timestamp_Diff(Current_Timestamp(), Current_Timestamp(), DAY)", {});
```

Let me know if you have further questions.

**Reply by rvalek** [reply]
This works. Thank you, sir!
